CMSgov/onemac
# Oy2 8865 timeout - Session expiry

Story: https://qmacbis.atlassian.net/browse/OY2-8865
Endpoint: https://d3mt5m03yif5u.cloudfront.net

### Details

When a user session times out and the user takes an action, the user receives a banner at the top.

“Your session time has expired. Please log back in to access OneMAC”.

### Changes

- Specific functionality changes, or
- A specific bug addressed in this PR


### Test Plan
Environment Setup:
==============
1. Login into AWS Cognito console (while on VPN)
2. Select option "User Pool" >> Click on "oy2-8865-time-out-user-pool" >> App Client >> Click on "More Details" on the right side detail pane
3. Set "Refresh Token Expiry" value to 0 days and 60 minutes (the available minimum value)
4. Set "Access Token Expiry" value to 0 days and 5 minutes
5. Set "Id Token Expiry" value to 0 days and 5 minutes
6. Click the "Save App Client Changes" button at the bottom of the page

Phone Number submission use case :
==========================
1. Login as the statesubmitteractive@cms.hhs.local
2. Navigate to "My Account" >> "Manage Profile" page
3. Click on the "Edit" button near the Phone Number >> Enter a phone number
4. Wait for at least one hour without any activity on the app (to let the session expire)
5. Click the "Apply" button.
6. You should see the Session expiry alert message at the top of the page

State Access Management use case :
==========================
1. Login as the statesubmitteractive@cms.hhs.local user
2. Navigate to "My Account" >> "Manage Profile" page
3. Add a State (Do not click the "Submit" button)
4. Wait for at least one hour without any activity on the app (to let the session expire)
5. Click the "Submit" button.
6. You should see the Session expiry alert message at the top of the page

State Access Management use case :
==========================
1. Login as the stateadminactive user
2. Navigate to "User Management"
3. Wait for at least one hour without any activity on the app (to let the session expire)
4. Click the three vertically aligned dots in one of the user rows (in the "Personal Actions" column) 
5. Click on "Grant Access" or "Deny Access" link
6. You should see the Session expiry alert message at the top of the page

Form Submission SPA ID validation use case :
===============================
1. Login as the statesubmitteractive@cms.hhs.local user
2. Navigate to Form Submission by clicking the "New Submission" button and go through the wizard to load the form
3. Wait for at least one hour without any activity on the app (to let the session expire)
4. Type in an ID in the SPA ID textbox
5. You should see the Session expiry alert message at the top of the page

Form Submission SPA ID validation use case :
===============================
1. Login as the statesubmitteractive@cms.hhs.local user
2. Navigate to Form Submission by clicking the "New Submission" button and go through the wizard to load the form
3. Type in an ID in the SPA ID textbox
4. Upload the required files
5. Wait for at least one hour without any activity on the app (to let the session expire)
6. Click the "Submit" button
7. You should see the Session expiry alert message at the top of the page
